File: quick.library/src/main/java/org/chris/quick/listener/OnClickListener2.java
package org.chris.quick.listener;

import android.util.Log;
import android.view.View;

import java.util.Calendar;

/**
 * Created by work on 2017/6/16.
 * 防止用户快速点击
 *
 * @author chris zou
 * @mail chrisSpringSmell@gmail.com
 */

public abstract class OnClickListener2 implements View.OnClickListener {

    long lastTime;
    public static final long intervalTime = 500;//间隔时间（毫秒）

    @Override
    public void onClick(View v) {
        long currentTime = Calendar.getInstance().getTimeInMillis();
        long temp = currentTime - lastTime;
        if (temp > intervalTime) {//间隔时间，成功触发
            onClick2(v);
            lastTime = currentTime;
            Log.e("点击", "触发，毫秒" + temp);
        }
        Log.e("点击差值", " 差值(毫秒) ：" + temp);
        Log.e("dfdf", "-----------------------------");
    }

    public abstract void onClick2(View view);
}

File: quick-component/src/main/java/org/quick/component/callback/OnClickListener2.kt
package org.quick.component.callback

import android.view.View

import org.quick.component.QuickAndroid

import java.util.Calendar

/**
 * Created by work on 2017/6/16.
 * 防止用户快速点击
 *
 * @author chris zou
 * @mail chrisSpringSmell@gmail.com
 */

abstract class OnClickListener2 : View.OnClickListener {

    private var lastTime: Long = 0

    override fun onClick(v: View) {
        val currentTime = Calendar.getInstance().timeInMillis
        val temp = currentTime - lastTime
        if (temp > QuickAndroid.defenseClickTime) {//间隔时间，成功触发
            onClick2(v)
            lastTime = currentTime
        }
    }

    abstract fun onClick2(view: View)
}
